feat(gateway): Comparison of db-less and decK formats (#6562)
Kong/developer.konghq.com: app/deck/dbless-vs-deck.md

Both formats use YAML and share some metadata fields, but they represent certain entities differently and aren't directly compatible.

## When to use each format

The following table summarizes when to use each format:

<!--vale off-->
{% table %}
columns:
  - title: Format
    key: format
  - title: Use case
    key: usecase
rows:
  - format: DB-less
    usecase: |
      Backup and restore, or configuring {{site.base_gateway}} in DB-less mode. Export with `kong config db_export` and import with `kong config db_import`. 
  - format: decK
    usecase: Human-authored, version-controlled {{site.base_gateway}} configuration for a database-backed deployment. Designed for manual editing and GitOps workflows.
{% endtable %}
<!--vale on-->

If you're using {{site.base_gateway}} in [DB-less mode](/gateway/db-less-mode/), you can't use decK for `sync`, `dump`, or similar operations because they require write access to the Admin API.

If you're running {{site.base_gateway}} with a database in [traditional](/gateway/traditional-mode/) or in [hybrid mode](/gateway/hybrid-mode/), decK is the better choice for managing entity configuration.

> The `kong config db_import` and `db_export` commands have several limitations:

Reviewer: ```suggestion
### {{site.base_gateway}} DB-less format limitations

The `kong config db_import` and `db_export` commands have several limitations:
```
This felt like a shift from the previous content comparing the two.

* **Cache invalidation**: `db_import` initializes a {{site.base_gateway}} database but isn't safe to run while existing nodes are running.
Changes aren't propagated to live nodes, so you'd have to restart all nodes manually.
decK applies changes through the Admin API, so all nodes receive updates automatically.
* **Deletions**: `db_import` can add and update entities, but it can't delete them.
It won't remove entities that exist in the database but are absent from the config file.
* **Direct database access**: `db_import` needs a direct connection to the {{site.base_gateway}} database, which may not be available in production networking environments.
* **Drift detection**: decK can compare the configuration in {{site.base_gateway}}'s database against your config file and report differences.
This is useful for CI pipelines or scheduled drift checks.
* **Readability**: `deck gateway dump` produces a more human-readable file than `db_export`.

> decK also has limitations to consider:

Reviewer: ```suggestion
### decK limitations

decK also has limitations to consider:
```

---

> title: "decK state file format vs {{site.base_gateway}} DB-less format"

Reviewer: ```suggestion
title: "Migrate from {{site.base_gateway}} DB-less format to decK state file format"
```
